com.unity.multiplayer.mlapi/Runtime/Core/RPCQueue/RPCQueueProcessing.cs

```diff
@@ -0,0 +1,227 @@
+        private static void RPCReceiveQueueProcessFlush()
+        {
+            bool AdvanceFrameHistory = false;
+            var rpcQueueManager = NetworkingManager.Singleton.RpcQueueManager;
+            if (rpcQueueManager != null)
+            {
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+                s_MLAPIRPCQueueProcess.Begin();
+#endif
+                var CurrentFrame = rpcQueueManager.GetCurrentFrame(QueueHistoryFrame.QueueFrameType.Inbound);
+                if (CurrentFrame != null)
+                {
+                    var currentQueueItem = CurrentFrame.GetFirstQueueItem();
+                    while (currentQueueItem.QueueItemType != RPCQueueManager.QueueItemType.None)
+                    {
+                        AdvanceFrameHistory = true;
+                        if (rpcQueueManager.IsLoopBack())
+                        {
+                            currentQueueItem.ItemStream.Position = 1;
+                        }
+
+                        NetworkingManager.InvokeRpc(currentQueueItem);
+                        ProfilerStatManager.rpcsQueueProc.Record();
+                        currentQueueItem = CurrentFrame.GetNextQueueItem();
+                    }
+
+                    //We call this to dispose of the shared stream writer and stream
+                    CurrentFrame.CloseQueue();
+                }
+
+                if (AdvanceFrameHistory)
+                {
+                    rpcQueueManager.AdvanceFrameHistory(QueueHistoryFrame.QueueFrameType.Inbound);
+                }
+            }
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+            s_MLAPIRPCQueueProcess.End();
+#endif
+        }
+
+        /// <summary>
+        /// ProcessSendQueue
+        /// Called to send both performance RPC and internal messages and then flush the outbound queue
+        /// </summary>
+        public void ProcessSendQueue()
+        {
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+            s_MLAPIRPCQueueSend.Begin();
+#endif
+
+            RPCQueueSendAndFlush();
+
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+            s_MLAPIRPCQueueSend.End();
+#endif
+            InternalMessagesSendAndFlush();
+        }
+
+        /// <summary>
+        ///  QueueInternalMLAPICommand
+        ///  Added this as an example of how to add internal messages to the outbound send queue
+        /// </summary>
+        /// <param name="queueItem">message queue item to add<</param>
+        public void QueueInternalMLAPICommand(FrameQueueItem queueItem)
+        {
+            internalMLAPISendQueue.Add(queueItem);
+        }
+
+        /// <summary>
+        /// Generic Sending Method for Internal Messages
+        /// TODO: Will need to open this up for discussion, but we will want to determine if this is how we want internal MLAPI command
+        /// messages to be sent.  We might want specific commands to occur during specific network update regions (see NetworkUpdate
+        /// </summary>
+        public void InternalMessagesSendAndFlush()
```

Comment:
👀

Reply:
Yeah, this should possibly be a completely new task that looks at how we handle the rest of the internal messages.  Specifically for creating and destroying objects (but there could be other related internal messages types I have yet to run across that need to have a specific order in how they are processed relative to RPCs).